enhancements/installer/openstack-upi.md

```diff
@@ -0,0 +1,316 @@
+  - "@luis5tb"
+  - "@wking"
+  - "@cuppett"
+approvers:
+  - TBD
+creation-date: 2019-09-10
+last-updated: 2019-12-03
+status: implementable
+---
+
+# OpenStack UPI
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria
+- [x] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The initial OpenStack support for OpenShift 4 was around the
+installer-provisioned (IPI) workflow. While this is the most
+convenient approach for the end-users of OpenStack, there are
+situations where it falls short. The goal of IPI is to create the
+infrastructure elements (network, VMs) needed for installing
+OpenShift. In order to reuse pre-existing pieces of infrastructure
+instead, users have to either adapt the [bare metal][baremetal-upi]
+flow or translate concepts from the AWS documentation.
+
+Similar to AWS and GCP, we want to provide documentation and tooling
+for OpenStack specifically -- using what OpenStack users are familiar
+with, and highlighting considerations specific to that platform.
+
+## Motivation
+
+In addition to the general use cases supported by UPI (e.g. control
+over the creation of the resources of the cloud provider, custom
+Ignition configuration), there are additional reasons for having an
+OpenStack-specific UPI.
+
+First, unlike for completely hosted solutions such as AWS or GCP there
+is no single OpenStack.
+
+Some deployments are public clouds akin to AWS while others are only
+available within a single organisation, possibly disconnected from the
+internet. Moreover, the actual services and configuration can vary
+greatly between deployments.
+
+The OpenStack UPI should be able to support scenarios such as these:
+
+* Lack of Swift Object Storage (which we use for serving the bootstrap
+  Ignition config as well as the registry storage)
+* No floating IP addresses (e.g. in deployments with provider
+  networks)
+* Disconnected from the internet
+* Desire to integrate all traffic with a load balancer or DNS that
+  already exists within the organisation running the OpenStack
+
+### Goals
+
+* OpenStack UPI documentation available under:
+https://github.com/openshift/installer/blob/master/docs/user/openstack/install_upi.md
+* Ansible playbooks for automating the OpenStack resource creation
+available under:
+https://github.com/openshift/installer/tree/master/upi/openstack
+* CI job executing these playbooks running wherever the AWS and GCP
+  jobs run
+* Optional documentation and playbook sections for Kuryr
+
+### Non-Goals
+
+Configurations and integrations that are not currently supported by
+the OpenStack IPI. For example:
+
+Things such as detailed steps for the provider network support, using
+Octavia or Designate for the load balancer / DNS, custom storage
+integration and so on can be added as optional sections once the
+initial UPI work is complete.
```

Comment:
hmm.. the goal of user-provisioned is to give users more freedom compared to installer-provisioned.

Can we scope this non-goals to specific use-cases that will not be supported. Or mark those use-cases not-implemented for now?

/cc @crawford @sdodson